packages/next/src/server/lib/experimental/typed-env.ts

```diff
@@ -0,0 +1,36 @@
+import type { Env } from '@next/env'
+import { existsSync } from 'node:fs'
+import { dirname, join } from 'node:path'
+import { mkdir, writeFile } from 'node:fs/promises'
+import { error } from '../../../build/output/log'
+
+export async function createEnvDefinitions(distDir: string, env: Env) {
+  const envKeys = Object.keys(env ?? {})
+  if (envKeys.length === 0) {
```

Comment:
should check if the file existed as well, if the env type file holds anything, need to be overriden or deleted